dll/win32/browseui/internettoolbar.cpp

```diff
@@ -1080,24 +1124,34 @@ HRESULT CInternetToolbar::IsBandVisible(int BandID)
     return (bandInfo.fStyle & RBBS_HIDDEN) ? S_FALSE : S_OK;
 }
 
-HRESULT CInternetToolbar::ToggleBandVisibility(int BandID)
+HRESULT CInternetToolbar::SetBandVisibility(int BandID, int Show)
 {
     int index = (int)SendMessage(fMainReBar, RB_IDTOINDEX, BandID, 0);
+    REBARBANDINFOW bandInfo = {sizeof(REBARBANDINFOW), RBBIM_STYLE | RBBIM_CHILD};
+    if (!SendMessage(fMainReBar, RB_GETBANDINFOW, index, (LPARAM)&bandInfo))
+        return E_FAIL;
 
-    REBARBANDINFOW bandInfo = {sizeof(REBARBANDINFOW), RBBIM_STYLE};
-    SendMessage(fMainReBar, RB_GETBANDINFOW, index, (LPARAM)&bandInfo);
-
-    if (bandInfo.fStyle & RBBS_HIDDEN)
+    if (Show < 0)
+        bandInfo.fStyle ^= RBBS_HIDDEN; // Toggle
+    else if (Show)
         bandInfo.fStyle &= ~RBBS_HIDDEN;
     else
         bandInfo.fStyle |= RBBS_HIDDEN;
 
+    bandInfo.fMask &= ~RBBIM_CHILD;
+    ::ShowWindow(bandInfo.hwndChild, (bandInfo.fStyle & RBBS_HIDDEN) ? SW_HIDE : SW_SHOW); // CORE-17236
     SendMessage(fMainReBar, RB_SETBANDINFOW, index, (LPARAM)&bandInfo);
 
     ReserveBorderSpace(0);
+    SetDirty();
     return S_OK;
 }
 
+HRESULT CInternetToolbar::ToggleBandVisibility(int BandID)
```

Comment:
Please keep in mind that you are now asking about code I did not write. All the existing code in CInternetToolbar uses int for the band id. Maybe because it compares it against -1 or other rebar return values, I don't know without studying the entire file.

Reply:
No problem, thanks for context explanation :)